tools/jolie2java/src/joliex/java/impl/JavaDocumentCreator.java

```
@@ -667,16 +667,18 @@ private void constructorCreate(StringBuilder stringBuilder, TypeDefinition type/
 
                     } else {
                         //native type
-                        String javaCode = javaNativeEquivalent.get(subType.nativeType());
-                        //String javaMethod = javaNativeMethod.get(subType.nativeType());
+                        String javaCode = javaNativeEquivalent.get(Utils.nativeType(subType));
+                        //String javaMethod = javaNativeMethod.get(Utils.nativeType(subType));
 
-                        if (((TypeDefinition) subType).cardinality().max() > 1) {
+                        if (subType.cardinality().max() > 1) {
                             stringBuilder.append("_" + subType.id() + "= new LinkedList<" + javaCode + ">();" + "\n");
                         }
                     }
 
+                } else if (subType instanceof TypeChoiceDefinition) {
+                    //can't initialize variable with several possible types
```

We should throw an error here.